astropy/units/format/fits.py

```diff
@@ -28,7 +28,10 @@ class Fits(generic.Generic):
     def _generate_unit_names():
         from astropy import units as u
 
-        names = {}
+        names = {
+            "Celsius": u.deg_C,
```

Comment:
Ah, that is such a simple solution, to just add them up front! But maybe add a comment along the lines of "Start with some units for which the standard unit name is different."